PublicWebAPI/Controllers/VenuesController.cs

```diff
@@ -0,0 +1,38 @@
+﻿using DataAccessLayer.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+using PublicWebAPI.Business.Dtos;
+using PublicWebAPI.Business.Services.Interfaces;
+
+namespace PublicWebAPI.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class VenuesController(IVenueService venueService) : Controller
+{
+    private readonly IVenueService _venueService = venueService ?? throw new ArgumentNullException(nameof(venueService));
+
+    [HttpGet]
+    public async Task<IActionResult> GetAll()
+    {
+        var venues = await _venueService.GetAllAsync();
+
+        return Ok(venues);
+    }
+
+    [HttpGet("{id:int}")]
+    public async Task<IActionResult> GetSections(int id)
+    {
+        SectionsToVenueDto sectionsToVenueDto;
+
+        try
+        {
+            sectionsToVenueDto = await _venueService.GetSectionsForVenue(id);
+        }
+        catch (RecordNotFoundException ex)
+        {
+            return BadRequest(ex.Message);
```

Comment:
It's not a good practice to show exception messages to the user, because they could expose some internal of your application for hackers.
It could be better to move all exception handling to global exception handler